xla/service/gpu/tests/gpu_fused_mha_test.cc

```diff
@@ -2770,7 +2770,7 @@ class FlashAttentionBMMScaleSoftmaxBMM : public MultiHeadedAttentionTest {
       Arg_0.1 = bf16[2,6,1024,64]{3,2,1,0} parameter(0), sharding={replicated}
       Arg_1.2 = bf16[2,6,64,1024]{3,2,1,0} parameter(1), sharding={replicated}
       dot.11 = bf16[2,6,1024,1024]{3,2,1,0} dot(Arg_0.1, Arg_1.2), lhs_batch_dims={0,1}, lhs_contracting_dims={3}, rhs_batch_dims={0,1}, rhs_contracting_dims={2}
-      constant.17 = bf16[] constant(2)
+      constant.17 = bf16[] constant(100)
```

Comment:
It looks like with this change, we get a test failure for FlashAttentionBMMScaleSoftmaxBMM.Flash_Attention_Training_BMM1_Softmax_BMM2_BF16

```
Mismatches in shape (bf16[2,6,1024,64], bf16[2,6,1024,64], bf16[2,6,64,1024], bf16[2,6,1024,64]) (3145728 elements):
Array at shape index {2}, 
Mismatch count 46025 (5.8524%) in shape bf16[2,6,64,1024] (786432 elements), abs bound 0.0025, rel bound 1e-05
Top relative error mismatches:
  actual     0.004272, expected    4.244e-05, index {1,2,17,590}, rel error      100, abs error  0.00424
  actual     -0.00415, expected   -4.125e-05, index {1,1,53,115}, rel error     99.9, abs error  0.00412
  actual     -0.00415, expected   -4.125e-05, index {0,3,57,689}, rel error     99.9, abs error  0.00412
  actual     -0.00415, expected   -4.125e-05, index {0,2,34,235}, rel error     99.9, abs error  0.00412
  actual     -0.00415, expected   -4.125e-05, index {0,2,3,432}, rel error     99.9, abs error  0.00412
```

Reply:
I guess 100 is too large? Maybe we just reduce it to sth like 20 or so.

Reply:
Can you reproduce the test failure? If yes, I guess you can try to find a value that prevents the failure, but still would cause a failure without the fixes from this PR?

Reply:
I rebased and I can reproduce this on A100. Magic number 37 is able to prevent the failure but will fail if the fix is removed.